LoRaEngine/modules/LoRaWanNetworkSrvModule/Logger/LogSink.cs

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan
+{
+    using System;
+    using System.Buffers;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Sockets;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Channels;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Extensions.Primitives;
+
+    public interface ILogSink
+    {
+        public LogLevel LogLevel { get; }
+        public void Log(LogLevel logLevel, string message);
+    }
+
+    public class LogSink : ILogSink
```

Comment:
What's the idea behind this `LogSink`? As far as I can see, it's only used once.

Reply:
It was used in two places but now I've switched to using for all subclasses with 39842d79de31986f47054d474d82468e8231cc7a. The idea behind it is that it's the base implementation.